soccerdata/fotmob.py

```diff
@@ -0,0 +1,434 @@
+            all_stats = game_data['content']['stats']['Periods']['All']['stats']
+            stats_dict = {i: all_stats[i]['title'] for i in range(len(all_stats))}
+
+            # if stat_type in stats_dict.values():
+            df_stats = pd.json_normalize(
+                all_stats, record_path='stats', meta=['title', 'key'], meta_prefix='cat_'
+            )
+            df_stat_type = df_stats[['title', 'stats', 'cat_title']].set_index('cat_title')
+
+            if stat_type not in stats_dict.values():
+                raise TypeError(f'Invalid argument: stat_type should be in {stats_dict.values()}')
+            else:
+                df_stat_type = df_stat_type.loc[df_stat_type.index == stat_type]
+            df_stat_type = (
+                df_stat_type.T.reset_index()
+                .set_axis(df_stat_type.T.reset_index().iloc[0], axis=1)
+                .iloc[1:]
+                .rename_axis(None, axis=1)
+            )
+            df_stat_type = df_stat_type.iloc[:, 2:]
+            for col in df_stat_type.columns:
+                df_single_stat = pd.DataFrame(df_stat_type[col])
+                df_stat_split = df_single_stat[col].apply(pd.Series)
+                df_single_stat = pd.concat([df_single_stat, df_stat_split], axis=1)
+                df_single_stat.drop(col, axis=1, inplace=True)
+                df_single_stat.rename(columns={0: 'Home ' + col, 1: 'Away ' + col}, inplace=True)
+                # Split percentage values into their own column
+                if df_single_stat['Home ' + col].dtypes == object:
+                    if (df_single_stat['Home ' + col].str.contains('%')).any():
+                        home_split_stats = df_single_stat['Home ' + col].str.split(
+                            r'[\(%]', expand=True
+                        )
+                        df_single_stat['Home ' + col] = int(home_split_stats[0].iloc[0])
+                        df_single_stat['Home ' + col + ' %'] = int(home_split_stats[1].iloc[0])
+                        away_split_stats = df_single_stat['Away ' + col].str.split(
+                            r'[\(%]', expand=True
+                        )
+                        df_single_stat['Away ' + col] = int(away_split_stats[0].iloc[0])
+                        df_single_stat['Away ' + col + ' %'] = int(away_split_stats[1].iloc[0])
+                stats.append(df_single_stat.reset_index())
+            # dfs = pd.concat(tmp_dfs, axis=1)
+            # dfs = dfs.T.drop_duplicates().T
+            # stats.append(tmp_dfs)
+            df_game = (
+                pd.concat(stats, axis=1).replace({'team': TEAMNAME_REPLACEMENTS}).sort_index()
+            )
+            # df_game = df_game.T.drop_duplicates().T
+            games.append(df_game)
+        df = (
+            pd.concat(games, ignore_index=True)
+            .replace({'team': TEAMNAME_REPLACEMENTS})
+            .set_index(['Matchweek', 'Date', 'Home Team', 'Home id', 'Away Team', 'Away id'])
```

Comment:
Indexes should be consistent across data sources using the following keys: ["league", "season", "game", "team", "player"]